File: javascript/reverse-linked-list.js
// O(n) time | O(1) space
const reverseList = (head) => {
    let prevNode = null;
    let currNode = head;
    
    while (currNode !== null) {
        let tempNode = currNode.next;
        currNode.next = prevNode;
        prevNode = currNode;
        currNode = tempNode;
    }
    
    return prevNode;
};

File: ruby/reverse-linked-list.rb
# O(n) time | O(n) space
def reverse_list(head)
    arr = []
    ptr = head
    
    while ptr != nil
        arr.push(ptr.val)
        ptr = ptr.next
    end
    
    nextPtr = nil
    arr.each do |val|
        node = ListNode.new(val, nextPtr)
        nextPtr = node
    end
    
    return nextPtr
end